streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -763,11 +763,19 @@ private KafkaStreams(final InternalTopologyBuilder internalTopologyBuilder,
             return thread;
         });
 
-        rocksDBMetricsRecordingTriggerThread = Executors.newSingleThreadScheduledExecutor(r -> {
-            final Thread thread = new Thread(r, clientId + "-RocksDBMetricsRecordingTrigger");
-            thread.setDaemon(true);
-            return thread;
-        });
+        rocksDBMetricsRecordingTriggerThread = maybeGetRocksDBMetricsRecordingTriggerThread(clientId, config);
+    }
+
+    private static ScheduledExecutorService maybeGetRocksDBMetricsRecordingTriggerThread(final String clientId,
+                                                                                         final StreamsConfig config) {
+        if (RecordingLevel.forName(config.getString(METRICS_RECORDING_LEVEL_CONFIG)) == RecordingLevel.DEBUG) {
```

Comment:
maybe instead to check that `rocksDBMetricsRecordingTriggerThread` is `null` instead? Here and elsewhere.

Reply:
Agreed!